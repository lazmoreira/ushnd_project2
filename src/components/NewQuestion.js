import { useState } from "react";
import { connect } from "react-redux";
import { useNavigate } from "react-router-dom";
import { handleAddQuestion } from "../actions/questions";

const NewQuestion = (props) => {
  const navigate = useNavigate();
  const { loggedUser, dispatch } = props;
  const [optionOne, setOptionOne] = useState("");
  const [optionTwo, setOptionTwo] = useState("");

  const handleChange = (e) => {
    e.target.id === "optionOne"
      ? setOptionOne(e.target.value)
      : setOptionTwo(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    dispatch(handleAddQuestion(optionOne, optionTwo, loggedUser));
    navigate("/");
  };

  return (
    <div>
      <h3>Create a new question</h3>
      <h1>Would You Rather</h1>
      <form onSubmit={handleSubmit}>
        <input
          type={"text"}
          placeholder={"Type the first option"}
          value={optionOne}
          className="option-input"
          maxLength={120}
          id="optionOne"
          onChange={handleChange}
        />
        <label htmlFor="optionOne">First option</label>
        <input
          type={"text"}
          placeholder={"Type the second option"}
          value={optionTwo}
          className="option-input"
          maxLength={120}
          id="optionTwo"
          onChange={handleChange}
        />
        <label htmlFor="optionTwo">Second option</label>
        <button
          className="btn"
          type="submit"
          disabled={optionOne === "" || optionTwo === ""}
        >
          Submit
        </button>
      </form>
    </div>
  );
};

const mapStateToProps = ({ loggedUser }) => ({
  loggedUser,
});

export default connect(mapStateToProps)(NewQuestion);
