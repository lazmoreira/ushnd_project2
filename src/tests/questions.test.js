import { render, screen } from "@testing-library/react";
import NewQuestion from "../components/NewQuestion";
import React from "react";
import { legacy_createStore as createStore } from "redux";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import reducer from "../reducers";
import middleware from "../middleware";

const store = createStore(reducer, middleware);

describe("NewQuestion", () => {
  it("will match snapshot", () => {
    const { component } = render(
      <MemoryRouter>
        <Provider store={store}>
          <NewQuestion />
        </Provider>
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });

  it("will verify if the fields are on the screen", () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <NewQuestion />
        </Provider>
      </MemoryRouter>
    );

    expect(screen.getByTestId("optionOne")).toBeInTheDocument();
    expect(screen.getByTestId("optionTwo")).toBeInTheDocument();
  });
});
