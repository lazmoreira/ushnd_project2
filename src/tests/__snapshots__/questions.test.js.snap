// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewQuestion will match snapshot 1`] = `undefined`;
